packages/vector_graphics/lib/vector_graphics.dart

```diff
@@ -1,7 +1,190 @@
+class VectorGraphicsAssetController extends VectorGraphicsController {
+  /// Create a new [VectorGraphicsAssetController].
+  ///
+  /// The default asset bundle can be acquired using [DefaultAssetBundle.of].
+  const VectorGraphicsAssetController({
+    required this.assetName,
+    this.packageName,
+    required this.assetBundle,
+  });
+
+  final String assetName;
+  final String? packageName;
+  final AssetBundle assetBundle;
+
+  @override
+  bool equivalent(VectorGraphicsController other) {
+    return other is VectorGraphicsAssetController &&
+        other.assetName == assetName &&
+        other.packageName == packageName &&
+        other.assetBundle == assetBundle;
+  }
+
+  @override
+  Future<ByteData> loadBytes() {
+    return assetBundle.load(assetName);
+  }
+}
+
+/// A controller for loading vector graphics data from over the network.
+class VectorGraphicsNetworkController extends VectorGraphicsController {
+  const VectorGraphicsNetworkController({
+    required this.url,
+    this.headers,
+    this.client,
+  });
+
+  final Map<String, String>? headers;
+  final Uri url;
+  final HttpClient? client;
+
+  @override
+  bool equivalent(VectorGraphicsController other) {
+    // This intentionally does not use [client].
+    return other is VectorGraphicsNetworkController &&
+        other.url == url &&
+        mapEquals(other.headers, headers);
+  }
+
+  @override
+  Future<ByteData> loadBytes() async {
+    final HttpClient currentClient = client ?? HttpClient();
+    final HttpClientRequest request = await currentClient.getUrl(url);
+    headers?.forEach(request.headers.add);
+
+    final HttpClientResponse response = await request.close();
+    if (response.statusCode != HttpStatus.ok) {
+      await response.drain<List<int>>(<int>[]);
+      throw Exception('Failed to load VectorGraphic: ${response.statusCode}');
+    }
+    final Uint8List bytes = await consolidateHttpClientResponseBytes(
+      response,
+    );
+    return bytes.buffer.asByteData();
+  }
+}
+
+class _RawVectorGraphicsWidget extends SingleChildRenderObjectWidget {
+  const _RawVectorGraphicsWidget({Key? key, required this.picture})
+      : super(key: key);
+
+  final ui.Picture picture;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderVectorGraphics(picture);
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, covariant _RenderVectorGraphics renderObject) {
+    renderObject.picture = picture;
+  }
+}
+
+class _RenderVectorGraphics extends RenderProxyBox {
+  _RenderVectorGraphics(this._picture);
+
+  ui.Picture get picture => _picture;
+  ui.Picture _picture;
+  set picture(ui.Picture value) {
+    if (identical(value, _picture)) {
+      return;
+    }
+    _picture = value;
+    markNeedsPaint();
+  }
+
+  @override
+  void paint(PaintingContext context, ui.Offset offset) {
+    context.canvas.translate(offset.dx, offset.dy);
```

Comment:
Maybe only do this if the offset != Offset.zero? I'm not sure we ever need this though. The offset is almost always going to be zero.

We're also missing whatever we'll need in here to scale the picture to the expected dimensions.

Reply:
Yeah this is really half assed right now, please do not seriously review this part of the code